File: Step 03 Arrays/Step 3.2 Medium/Array Leaders/Array Leaders.java
class Solution{
    //Function to find the leaders in the array.
    static ArrayList<Integer> leaders(int arr[], int n){
        // Your code here
        int s=0;
        ArrayList<Integer> ans=new ArrayList<>();
        for(int i=n-1;i>=0;i--){
            if(arr[i]>=s)
            ans.add(0,arr[i]);
            s=Math.max(s,arr[i]);
        }
        return ans;
    }
}

File: Step 03 Arrays/Step 3.2 Medium/Array Leaders/Array Leaders.cpp
    

class Solution{
    //Function to find the leaders in the array.
    public:
    vector<int> leaders(int a[], int n){
        // Code here
        vector<int> res;
        int leader = a[n-1];
        res.push_back(leader);
        for(int i=n-2;i>=0;--i) {
            if(leader <= a[i]) {
                leader = a[i];
                res.push_back(leader);
            }
        }
        reverse(res.begin(), res.end());
        return res;
    }
};